### DISCUSSION TITLE
Why are width and height required for next/image component?

### DISCUSSION THREAD

**Discussion by dotKuro** [vercel/next.js#19795] [Help]
The HTML img tag works perfectly fine without specifying a width nor a height. It just defaults to displaying the image in the original size. Why does the next/image component not work this way? For me it is really frustrating, because it is a really cool component with optimizations I'd like to use, but it seems that for seemingly no reason compatibility to the HTML img is broken. Neither using layout='fill' nor using 'auto' for width and height are solving the problem for me.

**Comment by deadcoder0904** [comment]
I think it's because of the Cumulative Layout Shift (CLS). If you specify `width` & `height` then when the image is loading, your content won't jump because you specified `width` & `height`.

**Reply by dotKuro** [reply]
Yes specifying width and height will result in rendering the picture. But I will have to look up the size of each picture manually and hardcode it per hand for every single picture for the website to look the same as with using img tag.

They are just incompatible in that which sense.

**Reply by deadcoder0904** [reply]
@dotKuro that's what I'm doing right now. Look up every image's dimensions manually & write them down. I think that will be a standard going forward as Google now takes into account CLS into ranking it seems.

`width` & `height` are back →

1. https://twitter.com/addyosmani/status/1169813271009886208?s=20
2. https://twitter.com/addyosmani/status/1276779799198007301?s=20

**Reply by trigaten** [reply]
I'm here with the same question, but I really hope hard coding is not the solution :)

**Comment by don-esteban** [comment]
As @deadcoder0904 wrote, there are good reasons to specify width/height for images. If you use HTML `<img>`, you should too! There are exceptions (fix container sizes, ...).

If you statically import images, tooling does the hard coding for you. If you load from external resources, see if they provide of sizing on the fly. Look to image hosting providers like Cloudinary. Another approach is using the "fill" property.

https://nextjs.org/docs/pages/api-reference/components/image

Good responsive image handling is not just placing an image tag somewhere at the page. With careful planning, the Next image component (together with other Next.js/React features) is a great help to build responsive sites even with hundreds of images on one page.

> Neither using layout='fill' nor using 'auto' for width and height are solving the problem for me.

Try to describe your issue and your current approach. Compare with a plain HTML, if you like. Perhaps a clean solution is closer as you expect.

Edit: Just see it now. A very old question. 😀
